jsx/Form.js

```diff
@@ -134,6 +134,131 @@ var FormElement = React.createClass({
   }
 });
 
+/**
+ * Datalist Component
+ * React wrapper for a searchable dropdown
+ */
+var DatalistElement = React.createClass({
+
+  propTypes: {
+    name: React.PropTypes.string.isRequired,
+    options: React.PropTypes.object.isRequired,
+    id: React.PropTypes.string,
+    strictDatalist: React.PropTypes.bool,
+    label: React.PropTypes.string,
+    value: React.PropTypes.oneOfType([
+      React.PropTypes.string,
+      React.PropTypes.array
+    ]),
+    class: React.PropTypes.string,
+    disabled: React.PropTypes.bool,
+    required: React.PropTypes.bool,
+    hasError: React.PropTypes.bool,
+    errorMessage: React.PropTypes.string,
+    onUserInput: React.PropTypes.func
+  },
+
+  getDefaultProps: function() {
+    return {
+      name: '',
+      options: {},
+      strictDatalist: false,
+      label: '',
+      value: undefined,
+      id: '',
+      class: '',
+      disabled: false,
+      required: false,
+      hasError: false,
+      errorMessage: 'The field is required!',
+      onUserInput: function() {
+        console.warn('onUserInput() callback is not set');
+      }
+    };
+  },
+
+  getKeyFromValue: function(value) {
+    var options = this.props.options;
+    var keyValue;
+    Object.keys(options).forEach(function(o) {
+      if (options[o] === value) {
+        keyValue = o;
+        return;
+      }
+    }, this);
+
+    return keyValue;
+  },
+
+  handleChange: function(e) {
+    var keyValue = this.getKeyFromValue(e.target.value);
+    this.props.onUserInput(this.props.name, keyValue);
+  },
+
+  handleBlur: function(e) {
+    // null out entry if not present in options in strict mode
+    if (this.props.strictDatalist) {
+      var value = e.target.value;
+      var options = this.props.options;
+      if (Object.values(options).indexOf(value) === -1) {
+        // null out both the hidden value as well as the input text
```

Comment:
I know this is a bit picky, but the values are being changed to empty strings, not `null`.